src/mark.js

```diff
@@ -385,6 +386,79 @@ class Mark { // eslint-disable-line no-unused-vars
     }
 
     /**
+     * Check if a value is a number
+     * @param {number|string} value - the value to check;
+     * numeric strings allowed
+     * @return {boolean}
+     * @access protected
+     */
+    isNumeric(value) {
+        // http://stackoverflow.com/a/16655847/145346
+        // eslint-disable-next-line eqeqeq
+        return Number(parseFloat(value)) == value;
+    }
+
+    /**
+     * @typedef Mark~setOfRanges
+     * @type {object[]}
+     * @property {number} start - The start position within the composite value
+     * @property {number} end - The end position within the composite value
+     * @property {number} len - The length of the string to mark within the
+     * composite value. If set, this value will override the end value
+     */
+    /**
+     * Returns a processed list of integer offset indexes that do not overlap
+     * each other, and remove any string values or additional elements
+     * @param {Mark~setOfRanges} array - unprocessed raw array
+     * @return {Mark~setOfRanges} - processed array with any invalid entries
+     * removed
+     * @throws Will throw an error if an array of objects is not passed
+     * @access protected
+     */
+    checkRanges(array) {
+        // start, end or length indexes are included in an array of objects
+        // [{start: 0, end: 1}, {start: 4, len: 5}]
+        // quick validity check of the first entry only
+        if (
+            !Array.isArray(array) ||
+            Object.prototype.toString.call( array[0] ) !== "[object Object]"
+        ) {
+            throw new Error("markRange() will only accept an array of objects");
+        }
```

Comment:
To me, it makes sense to use `start` and `end`.... it is closer to how a range is defined by W3C, e.g. for a [range type of input](https://www.w3.org/wiki/HTML/Elements/input/range), you set a  `min` and `max`:

```html
<input type="range" min="1" max="10">
``` 

But if third party applications return a `start` and `length`, accommodating the plugin to work with these types of values makes it easier on the developer. I know you can pass the function a `start` and `start + length` which is what I used in the [initial demo I shared](https://jsfiddle.net/Mottie/whkmkpfm/) (this demo no longer works since this branch was updated).

I don't really see any cons for including both since there is only a minimal amount of coding requires to support both.